### ISSUE TITLE
where is cnmf.py located

### ISSUE THREAD

**Issue by ryao-mdanderson** [dylkot/cNMF#15]
Dear author,

 Does conda have to be upgraded before install the virtual environment? I did not. 
 
 Here is my commands
 $ conda create -n cnmf_env2 --yes --channel bioconda --channel conda-forge --channel defaults python==3.7 fastcluster==1.1.26 matplotlib==3.3.2 numpy==1.19.2 palettable==3.3.0 pandas==1.1.3 scipy==1.5.2 scikit-learn==0.23.2 pyyaml==5.3.1 scanpy==1.6.0 parallel
Collecting package metadata (repodata.json): done
Solving environment: done
 ........
 ==> WARNING: A newer version of conda exists. <==
  current version: 4.9.2
  latest version: 4.10.1

Please update conda by running

    $ conda update -n base conda

## Package Plan ##

  environment location: /rsrch3/home/itops/ryao/.conda/envs/cnmf_env2

  added / updated specs:
    - fastcluster==1.1.26
    - matplotlib==3.3.2
    - numpy==1.19.2
    - palettable==3.3.0
    - pandas==1.1.3
    - parallel
    - python==3.7
    - pyyaml==5.3.1
    - scanpy==1.6.0
    - scikit-learn==0.23.2
    - scipy==1.5.2

Installation goes without an error message, the packages installed in my local .conda environment.

after activate the conda environment,
(cnmf_env2) [ryao@ldragon2 cnmf_env2]$ ls -lrt /rsrch3/home/itops/ryao/.conda/envs/cnmf_env2
total 205
drwx------  4 ryao rists  4096 May 28 15:09 man
drwx------  3 ryao rists  4096 May 28 15:09 ssl
drwx------  3 ryao rists  4096 May 28 15:09 var
drwx------  2 ryao rists  4096 May 28 15:09 compiler_compat
drwx------  2 ryao rists  4096 May 28 15:09 lib64
drwx------  6 ryao rists  4096 May 28 15:09 etc
drwx------  3 ryao rists  4096 May 28 15:09 doc
drwx------ 36 ryao rists 16384 May 28 15:09 include
drwx------ 15 ryao rists  4096 May 28 15:09 plugins
drwx------ 28 ryao rists 32768 May 28 15:10 lib
drwx------  4 ryao rists  4096 May 28 15:10 libexec
drwx------ 97 ryao rists  4096 May 28 15:10 mkspecs
drwx------  2 ryao rists  4096 May 28 15:10 phrasebooks
drwx------ 14 ryao rists  4096 May 28 15:10 qml
drwx------  2 ryao rists 16384 May 28 15:10 translations
drwx------ 27 ryao rists  4096 May 28 15:10 share
drwx------  2 ryao rists 16384 May 28 15:10 bin
drwx------  2 ryao rists 16384 May 28 15:10 conda-meta

(cnmf_env2) [ryao@ldragon2 cnmf_env2]$ find . -name cnmf.py
but I can't find the file, thus the first step fails:

(cnmf_env2) [ryao@ldragon2 ~]$ python cnmf.py --help
(null): can't open file 'cnmf.py': [Errno 2] No such file or directory

Please advise, Thank you!
Rong Yao

**Comment by fursham-h** [comment]
You would need to clone the repo to your local directory. Conda sets up the dependencies for cNMF but not the script itself.

**Comment by ryao-mdanderson** [comment]
Dear Author,

Thanks for your reply. Right, I git clone the the repo in the same location of the package environment to locate the cnmf.py. 

Followed REDAME.md, I download example_simulated_data; so this is the directory structure : 
$ pwd
/rsrch3/home/itops/ryao/.conda/envs/cnmf_env2
$ ls -l
total 8143
drwx------  2 ryao rists   16384 May 28 15:10 bin
drwx------  5 ryao rists    4096 Jun  1 09:53 cNMF
drwx------  2 ryao rists    4096 May 28 15:09 compiler_compat
drwx------  2 ryao rists   16384 May 28 15:10 conda-meta
drwx------  3 ryao rists    4096 May 28 15:09 doc
drwx------  6 ryao rists    4096 May 28 15:09 etc
-rw-r--r--  1 ryao rists 8115726 Jun  1 10:11 example_simulated_data.tar.gz
drwx------ 36 ryao rists   16384 May 28 15:09 include
drwx------ 28 ryao rists   32768 May 28 15:10 lib
drwx------  2 ryao rists    4096 May 28 15:09 lib64
drwx------  4 ryao rists    4096 May 28 15:10 libexec
drwx------  4 ryao rists    4096 May 28 15:09 man
drwx------ 97 ryao rists    4096 May 28 15:10 mkspecs
drwx------  2 ryao rists    4096 May 28 15:10 phrasebooks
drwx------ 15 ryao rists    4096 May 28 15:09 plugins
drwx------ 14 ryao rists    4096 May 28 15:10 qml
drwx------ 27 ryao rists    4096 May 28 15:10 share
drwx------  3 ryao rists    4096 Jun  1 10:17 simulated_example_data
drwx------  3 ryao rists    4096 May 28 15:09 ssl
drwx------  2 ryao rists   16384 May 28 15:10 translations
drwx------  3 ryao rists    4096 May 28 15:09 var
(cnmf_env2) [ryao@ldragon2 cnmf_env2]$ ls -l simulated_example_data/
total 43521
-rwx------ 1 ryao rists  4684305 Oct 28  2019 counts.npz
drwx------ 3 ryao rists     4096 Jun  1 10:17 example_cNMF
-rwx------ 1 ryao rists 37157256 Oct 28  2019 filtered_counts.txt
-rwx------ 1 ryao rists   125956 Oct 28  2019 groundtruth_cellparams.txt
-rwx------ 1 ryao rists  2391839 Oct 28  2019 groundtruth_geneparams.txt
-rwx------ 1 ryao rists   116536 Oct 28  2019 tsne.txt

I have X11 forwarding set up and run the following command: I did not find 
$ python cNMF/cnmf.py prepare --output-dir ./simulated_example_data/ --name example_cNMF -c ./simulated_example_data/filtered_counts.txt -k 5 6 7 8 9 10 11 12 13 --n-iter 100 --total-workers 1 --seed 14 --numgenes 2000
-- output
/home/ryao/.local/lib/python3.7/site-packages/anndata/_core/anndata.py:21: FutureWarning: pandas.core.index is deprecated and will be removed in a future version.  The public classes are available in the top-level namespace.
  from pandas.core.index import RangeIndex
/home/ryao/.local/lib/python3.7/site-packages/anndata/_core/anndata.py:1056: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
/home/ryao/.local/lib/python3.7/site-packages/scanpy/preprocessing/_simple.py:912: UserWarning: Revieved a view of an AnnData. Making a copy.
  view_to_actual(adata)

after that, I saw update in simulated_example_data/ directory:
$ ls -lrt simulated_example_data/
total 43521
-rwx------ 1 ryao rists  4684305 Oct 28  2019 counts.npz
-rwx------ 1 ryao rists  2391839 Oct 28  2019 groundtruth_geneparams.txt
-rwx------ 1 ryao rists   125956 Oct 28  2019 groundtruth_cellparams.txt
-rwx------ 1 ryao rists 37157256 Oct 28  2019 filtered_counts.txt
-rwx------ 1 ryao rists   116536 Oct 28  2019 tsne.txt
drwx------ 3 ryao rists     4096 Jun  1 10:17 example_cNMF
(cnmf_env2) [ryao@cdragon305 cnmf_env2]$ ls -lrt simulated_example_data/example_cNMF/
total 65
-rwx------ 1 ryao rists 17794 Jun  1 10:17 example_cNMF.overdispersed_genes.txt
drwx------ 2 ryao rists  4096 Jun  1 10:17 cnmf_tmp

Does this look as expected? I am a systems admin and help to set up the application on HPC cluster; at this point, is it Okay to allow our computational scientists to further try the example tutorial data?

Thanks for your help.
Rong
